lib/pltr/v2/actions/__tests__/ui.test.js

```diff
@@ -224,3 +250,178 @@ describe('cardDialog', () => {
     })
   })
 })
+
+describe('bookDialog', () => {
+  const store = initialStore()
+  const initialState = removeSystemKeys(store.getState().present)
+
+  describe('given the initial state store', () => {
+    const bookId = bookDialogBookIdSelector(initialState)
+    const isOpen = isBookDialogVisibleSelector(initialState)
+    it('should produce initial state store', () => {
+      expect(initialState.ui.bookDialog).toMatchObject(initialBookDialogState)
+    })
+
+    it('should have `null` as initial values for cardId lineId and beatId', () => {
+      expect(bookId).toBeNull()
+    })
+
+    it('should have false as initial value for isOpen', () => {
+      expect(isOpen).toBeFalsy()
+    })
+  })
+
+  describe('given bookDialog actions are dispatched', () => {
+    describe('and openNewBookDialog is dispatched', () => {
+      const nextBookNumber = allBookIdsSelector(initialState).length + 1
+      store.dispatch(openNewBookDialog())
+      const previousState = store.getState().present
+      const bookDialogBookId = bookDialogBookIdSelector(previousState)
+      const bookNumber = bookNumberSelector(previousState)
+      const isOpen = isBookDialogVisibleSelector(previousState)
+      const initialTotalNumberOfBooks = allBookIdsSelector(previousState).length
+
+      it('should open the bookDialog', () => {
+        expect(isOpen).toBeTruthy()
+      })
+
+      it('should not have a bookId', () => {
+        expect(bookDialogBookId).toBeNull()
+      })
+
+      it('should change the bookNumber equal to the param passed', () => {
+        expect(bookNumber).toEqual(nextBookNumber)
+      })
+
+      describe('given addBook is dispatched with completely blank fields', () => {
+        store.dispatch(addBook())
+        const presentState = store.getState().present
+        const allBooks = allBooksSelector(presentState)
+        const totalBooks = allBookIdsSelector(presentState).length
+
+        it('should work as before and save the book without passing any params', () => {
+          expect(totalBooks).toBeGreaterThan(initialTotalNumberOfBooks)
+
+          const book = mapValues(allBooks, (value, key, obj) => {
+            if (typeof value == 'object' && !value.title) {
+              return value
+            }
+            return undefined
+          })
+          const filteredBook = Object.keys(book).filter((b) => !!book[b])
+          expect(filteredBook.title).toBeUndefined()
+          expect(filteredBook.theme).toBeUndefined()
+          expect(filteredBook.title).toBeUndefined()
```

Comment:
I think this and the next line are typos(?)

Reply:
Nearly there!  It still checks `title` three times :wink:

Reply:
@Quiescent sorry 🥲 thank you for your patience checking

Reply:
No problem! :)